contrib/Dockerfile

```diff
@@ -1,10 +1,48 @@
-# This file is used to add the nightly Dgraph binaries and assets to Dgraph base
-# image.
+# This file builds the dgraph binary and dgraph docker image.
 
-# This gets built as part of release.sh. Must be run from /tmp/build, with the linux binaries
-# already built and placed there.
+# This gets built as part of our CD pipeline. Must be run through Makefile.
 
-FROM ubuntu:20.04
+FROM --platform=linux/amd64 ubuntu:20.04 AS build
+
+RUN mkdir /dgraph
+WORKDIR /dgraph
+
+COPY . ./
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    curl \
+    ca-certificates
+
+RUN GO_VERSION=$({ [ -f .go-version ] && cat .go-version; }) \
+    && curl -O -L "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" \
+    && tar -C /usr/local/ -xzf go${GO_VERSION}.linux-amd64.tar.gz \
+    && rm -f go${GO_VERSION}.linux-amd64.tar.gz
+
+ENV PATH=$PATH:/usr/local/go/bin:/root/go/bin \
+    GOPATH=/root/go
+
+# save time by caching this layer
+RUN cd dgraph && make jemalloc
+
+ARG BUILD
+ARG BUILD_CODENAME
+ARG BUILD_DATE
+ARG BUILD_BRANCH
+ARG BUILD_VERSION
+
+RUN BUILD=$BUILD \
+    BUILD_CODENAME=$BUILD_CODENAME \
+    BUILD_DATE=$BUILD_DATE \
+    BUILD_BRANCH=$BUILD_BRANCH \
+    BUILD_VERSION=$BUILD_VERSION \
+    make dgraph
+
+RUN mkdir bin && mv dgraph/dgraph bin
+
+# -------------------------------------------------------------------------------
+
+FROM --platform=linux/amd64 ubuntu:20.04
```

Comment:
this flag is needed to make this work on apple m1; otherwise it will default to linux/arm64

Reply:
we should just add this comment to Dockerfile itself

Reply:
I don't follow?  this flag changes the base image that is used.

Reply:
I should mention that this is an issue since we will eventually want to give first class support to linux/arm64 and these kinds of explicit parameters preclude that.

Reply:
the comment that you made here on GitHub, I am suggesting that the same comment should be present in line 45 of the Dockerfile.